server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -155,16 +169,19 @@ protected void parseCreateField(ParseContext context) throws IOException {
 
     @Override
     public final void parse(ParseContext context) throws IOException {
-      parser.parse(context.parser(), v -> index(context, v), e -> {
-          if (ignoreMalformed()) {
-              context.addIgnoredField(fieldType().name());
-          } else {
-              throw new MapperParsingException(
-                  "Failed to parse field [" + fieldType().name() + "] of type [" + contentType() + "]",
-                  e
-              );
-          }
-      });
+        if (hasScript) {
+            throw new MapperParsingException("failed to parse field [" + fieldType().name() + "] of type + " + contentType() + "]",
+                new IllegalArgumentException("Cannot index data directly into a field with a [script] parameter"));
```

Comment:
grr I forgot about the mappers that override parse when I added this to the base class. Is it worth extracting the check to a common method so that it can be called from here?

Reply:
I want to rework parse/parseCreateField anyway, as it's very confusing that we have these two methods with the same signature that do almost exactly the same thing.  Can we consider this a band-aid until we tidy up in a followup?

Reply:
sure, anyways the test is unified and tests the error message, so we ensure that it is always the same one